cdap-gateway/src/main/java/co/cask/cdap/gateway/router/NettyRouter.java

```diff
@@ -201,11 +202,7 @@ protected void shutDown() throws Exception {
   protected Executor executor(final State state) {
     final AtomicInteger id = new AtomicInteger();
     //noinspection NullableProblems
-    final Thread.UncaughtExceptionHandler h = new Thread.UncaughtExceptionHandler() {
-      @Override
-      public void uncaughtException(Thread t, Throwable e) {
-      }
-    };
+    final Thread.UncaughtExceptionHandler h = new UncaughtExceptionHandler();
```

Comment:
The existing code is pretty stupid. The executor is only used for this abstract idle service. I think it's better to remove the setting of uncaught exception handler per thread here, and then set the default uncaught exception handler in `RouterMain`.